tests/sqlparser_sqlite.rs

```diff
@@ -290,6 +290,31 @@ fn parse_create_table_with_strict() {
     }
 }
 
+#[test]
+fn parse_window_function_with_filter() {
+    let sql = "SELECT max(x) FILTER (WHERE y) OVER () FROM t";
```

Comment:
I wonder if you  can add another test with for a window function (aka that has an `OVER` clause) but that is not an aggregate. For example:

```sql
SELECT row_number() FILTER (WHERE y) OVER () FROM t
```

I tried it in postgres and found the syntax is parsed but the functionality is not implemented

```
ERROR:  FILTER is not implemented for non-aggregate window functions
LINE 1: SELECT row_number() FILTER (WHERE column1 > 1) OVER () FROM ...
```

Reply:
done in https://github.com/sqlparser-rs/sqlparser-rs/pull/1007/commits/dd51436086f0123abe7a31bb24d5515b936a091d